mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -99,8 +104,9 @@ class _ChatRequest(pydantic.BaseModel, extra="forbid"):
 
 class _ChatChoice(pydantic.BaseModel, extra="forbid"):
```

Comment:
I'd define a different class for delta.

Reply:
@WeichenXu123 you can check what the openai python SDK does as a refrence.

Reply:
I tested with

```
from openai import OpenAI
import os

client = OpenAI(api_key=os.environ.get("OPENAI_API_KEY", "..."))

response = client.chat.completions.create(
    model='gpt-3.5-turbo',
    messages=[
        {'role': 'user', 'content': 'Count to 5. E.g., 1, 2, 3, ...'}
    ],
    temperature=0,
    stream=<True or False>,
)
```

If stream = false,
we get
```
ChatCompletion(id='chatcmpl-97LQf70aMgTPLMsWtIESQCxI2RhnQ', choices=[Choice(finish_reason='stop', index=0, logprobs=None, message=ChatCompletionMessage(content='1, 2, 3, 4, 5', role='assistant', function_call=None, tool_calls=None))], created=1711537697, model='gpt-3.5-turbo-0125', object='chat.completion', system_fingerprint='fp_3bc1b5746c', usage=CompletionUsage(completion_tokens=13, prompt_tokens=25, total_tokens=38))
```

if stream = true,
the first chunk we get is:

```
ChatCompletionChunk(id='chatcmpl-97LRYGtlDoMJvcMmT8rPpDLYAFLeS', choices=[Choice(delta=ChoiceDelta(content='', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1711537752, model='gpt-3.5-turbo-0125', object='chat.completion.chunk', system_fingerprint='fp_3bc1b5746c')
```

So seemingly they are using the same class for `choice`

Reply:
Added a new class `_ChatDeltaMessage`